xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/ExportAction.java

```diff
@@ -94,6 +111,43 @@ public String render(XWikiContext context) throws XWikiException
         return defaultPage;
     }
 
+    private class ExportArguments {
+        private String[] checkedPages;
+        private String[] uncheckedPages;
+        private boolean otherPages;
+        private String name;
+        private String description;
+
+        ExportArguments(XWikiContext context)
+        {
+            XWikiRequest request = context.getRequest();
+
+            this.description = request.get("description");
+
+            this.name = request.get("name");
+            if (StringUtils.isBlank(name)) {
+                this.name = context.getDoc().getFullName();
+            }
+
+            String[] pagesParam = request.getParameterValues("pages");
+            String checkedPagesParam = request.get("checked-pages");
+
+            if (!StringUtils.isEmpty(checkedPagesParam)) {
+                this.checkedPages = checkedPagesParam.split("&");
```

Comment:
As @mflorea pointed out in a previous comment this code seems to be impossible to have a page name containing `&` in that list. If you insist on inventing your own syntax you will need escaping support in this syntax because in XWiki there is no forbidden characters in names.

Reply:
The names of pages are encoded in the exporter.js, see: https://github.com/xwiki/xwiki-platform/pull/855/files#diff-732756cd30b04d597a99718c625c573aR246

So the "&" which might exist in the filename are already encoded.
Now as I answered in your previous comment I cannot use the `getParamValues` as it would involve create as many input fields as I have checked pages: if you have any other advice to not use this syntax, I take it.

Reply:
The code would be much easier to understand if the decode was done here. So that checkedPages actually contain a list of pages :)

Reply:
Indeed. I'll fix that.